### ISSUE TITLE
KeyError: 'site_damp' after Conflicting integration found

### ISSUE THREAD

**Issue by BJReplay** [BJReplay/ha-solcast-solar#221]
## Describe the bug

After forcing a conflicting integration to be found (by copying the install directory - `cp -r solcast_solar salcast_solar` hitting Configure results in an error:

## To Reproduce

1. In a Dev Container environment, download the integration
2. Copy the install directory in custom_components prior to installing the integration and prior to restarting HA
3. Attempt to install the integration
4. Note the Conflicting integration ERROR in the logs
5. Delete the extra directory in the custom_components directory e.g. `rm -rf salcast_solar` 
6. Attempt to `CONFIGURE` the partially installed integration.
7. Note the horror (**logged exception below displayed to user**)
8. Restart HA Dev Container
9. Note that the integration completes setup as expected.
10. Note the horror is replaced with goodness.

## Expected behavior

Less horror.

Really, I guess what I think should happen is that the error message being logged needs to indicate that the extra directory needs to be removed, **_and HA needs to be restarted_** before the integration can continue.

## Screenshots

![image](https://github.com/user-attachments/assets/37428b32-fd07-40d2-8d0e-b3efd3f18dbc)

## Logs

I confirm:

- [X] I have attached **debug** logs
- [ ] I have embedded **debug** logs in the issue description (enclosed in tick marks ``` for proper formatting)
- [ ] Confirmed **debug** logs are not required for this issue

## Solcast Integration Version

- Integration Version v4.2.6

## Desktop (please complete the following information)

- OS: Dev Container

## Logs

```
2024-11-23 09:43:54.629 ERROR (MainThread) [custom_components.salcast_solar] Conflicting integration found in solcast_solar, code owners: ['@BJReplay', '@autoSteve']
2024-11-23 09:44:33.583 ERROR (MainThread) [aiohttp.server] Error handling request
Traceback (most recent call last):
  File "/home/vscode/.local/ha-venv/lib/python3.12/site-packages/aiohttp/web_protocol.py", line 477, in _handle_request
    resp = await request_handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/ha-venv/lib/python3.12/site-packages/aiohttp/web_app.py", line 559, in _handle
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/ha-venv/lib/python3.12/site-packages/aiohttp/web_middlewares.py", line 117, in impl
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/core/homeassistant/components/http/security_filter.py", line 92, in security_filter_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/core/homeassistant/components/http/forwarded.py", line 83, in forwarded_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/core/homeassistant/components/http/request_context.py", line 26, in request_context_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/core/homeassistant/components/http/ban.py", line 86, in ban_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/core/homeassistant/components/http/auth.py", line 242, in auth_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/core/homeassistant/components/http/headers.py", line 32, in headers_middleware
    response = await handler(request)
               ^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/core/homeassistant/helpers/http.py", line 73, in handle
    result = await handler(request, **request.match_info)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/core/homeassistant/components/http/decorators.py", line 81, in with_admin
    return await func(self, request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/core/homeassistant/components/config/config_entries.py", line 262, in post
    return await super().post(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/core/homeassistant/components/http/data_validator.py", line 74, in wrapper
    return await method(view, request, data, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/core/homeassistant/helpers/data_entry_flow.py", line 77, in post
    return await self._post_impl(request, data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/core/homeassistant/helpers/data_entry_flow.py", line 84, in _post_impl
    result = await self._flow_mgr.async_init(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/core/homeassistant/data_entry_flow.py", line 342, in async_init
    result = await self._async_handle_step(flow, flow.init_step, data)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/core/homeassistant/data_entry_flow.py", line 517, in _async_handle_step
    result: _FlowResultT = await getattr(flow, method)(user_input)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspaces/core/config/custom_components/salcast_solar/config_flow.py", line 282, in async_step_init
KeyError: 'site_damp'

```

**Comment by autoSteve** [comment]
That is a thing of beauty.

Manufactured beauty, but logged beauty nonetheless.

I believe it is manufactured because the "salcast_solar" identified integration was never completely set up in HA, yeah? The screen shot would not indicate that the conflict is coming from "solcast_solar" in that case, rather it should indicate "salcast_solar" is conflicting.

If both integrations were doing the check then this screen shot could happen, with salcast getting in first. So a better test would be v4.2.5 vs v4.2.6.

All that said, I reckon it's a moot point, and I am doing the check for a conflict in entirely the wrong place. It **_has_** to be earlier.

To avoid all of this logged beauty I reckon that it will need to be in the config flow to prevent the integration even being partially set up. Right after this in `config_flow.py`...

```
        if self._async_current_entries():
            return self.async_abort(reason="single_instance_allowed")
```

**Comment by autoSteve** [comment]
And in config flow it'd go a little somethin' like this...

![image](https://github.com/user-attachments/assets/6a263b5f-252a-4167-8391-97f0755e3835)

No logs harmed.

```
[custom_components.solcast_solar.config_flow] Conflicting integration found in aolcast_solar_old, code owners: ['@Oziee']
```

Right now only the presence of the _files_ and their content is tested for, not whether the foul and offending competitor is actually running. I should be able to lookup the current domains. That is on my to-ponder-further list, as is getting a non-blocking async call to happen in `config_flow.py`.

**Comment by autoSteve** [comment]
Largely un-tested move of conflict check.

**Comment by autoSteve** [comment]
Better tested, but imperfect.

I set up v4.2.0 as `aolcast_solar_old`, and configured against the sim key `1`. Running.

Tried to set up `solcast_solar`, which barfed with "Conflicting integration found: aolcast_solar_old". All good. It checks for services being exposed by a competitor, so must be running and not just the files there.

Deleted `aolcast_solar_old` integration, but left all the files downloaded and in place. Conflict check failed on install of `solcast_solar` when it should not. Bummer. Restart HA. Then all is good with that install.

It wants tightening up. I believe the `dict` containing services may not be cleaned up on integration deletion, so I continue to look for a definitive kill.

But...

This is a scenario that requires the install folder to be modified, `manifest.json` to be modified, _and_ `const.py` to be modified. If these three things are not done then no competing integration. A truly perverted scenario.

I can't help but think there is something far more simpler to the whole, "Yeah, dude, like Oziee and this one were totally running together..." thing.

Zero evidence. Zero confidence.

**Comment by BJReplay** [comment]
> and I am doing the check for a conflict in entirely the wrong place. It **_has_** to be earlier

This, I feel, is true.

Even though my test was entirely manufactured, it seems like the only sort of thing that could possibly have led to the claims - I'm gonna call them that without github tagging people - of duplicate integrations running that never followed through with evidence.

**Comment by BJReplay** [comment]
Will be closed in #222.  Open until merged.
